lib/notifiers/notifiers.go

```diff
@@ -214,19 +237,38 @@ func Main(notifier Notifier) error {
 	if err != nil {
 		return fmt.Errorf("failed to get config from GCS: %w", err)
 	}
+
 	if err := validateConfig(cfg); err != nil {
 		return fmt.Errorf("got invalid config from path %q: %w", cfgPath, err)
 	}
 	log.V(2).Infof("got config from GCS (%q): %+v\n", cfgPath, cfg)
-
+	// Get Template Logic
+	tmpl := ""
+	if cfg.Spec.Notification.Template != nil {
+		if _, ok := allowedTemplateTypes[cfg.Spec.Notification.Template.Type]; !ok {
+			return fmt.Errorf("got invalid Template Type: %v", cfg.Spec.Notification.Template.Type)
+		}
+		if cfg.Spec.Notification.Template.URI != "" {
+			tmpl, err = getGCSTemplate(ctx, &actualGCSReaderFactory{sc}, cfg.Spec.Notification.Template.URI)
+			if err != nil {
+				return fmt.Errorf("failed to get template from GCS: %w", err)
+			}
+		} else {
+			tmpl = cfg.Spec.Notification.Template.Content
```

Comment:
what about a unit test for this use case?

Reply:
extracted the logic from that section into a testable function and added tests for it